src/java/org/jivesoftware/openfire/muc/spi/LocalMUCRoom.java

```diff
@@ -373,7 +373,7 @@ public LocalMUCRoom() {
         this.canChangeNickname = MUCPersistenceManager.getBooleanProperty(mucService.getServiceName(), "room.canChangeNickname", true);
         this.registrationEnabled = MUCPersistenceManager.getBooleanProperty(mucService.getServiceName(), "room.registrationEnabled", true);
         // TODO Allow to set the history strategy from the configuration form?
-        roomHistory = new MUCRoomHistory(this, new HistoryStrategy(mucService.getHistoryStrategy()));
+        roomHistory = new MUCRoomHistory(this, mucService.getHistoryStrategy());

```

Comment:
I think each room needs its own instance, at least for the cached room subject property. Additional refactoring will likely be needed to make the history instance shareable across rooms within the MUC service.